File: src/main/java/org/mitallast/queue/common/component/ComponentModule.java
package org.mitallast.queue.common.component;

import com.google.inject.AbstractModule;
import com.typesafe.config.Config;

public class ComponentModule extends AbstractModule {

    private final Config config;

    public ComponentModule(Config config) {
        this.config = config;
    }

    @Override
    protected void configure() {
        LifecycleService lifecycleService = new LifecycleService();
        bind(Config.class).toInstance(config);
        bind(LifecycleService.class).toInstance(lifecycleService);
        bindListener(new LifecycleMatcher(), lifecycleService);
    }
}

File: src/main/java/org/mitallast/queue/common/component/ComponentModule.kt
package org.mitallast.queue.common.component

import com.google.inject.AbstractModule
import com.typesafe.config.Config
import org.mitallast.queue.common.logging.LoggingService

class ComponentModule(private val config: Config, private val logging: LoggingService) : AbstractModule() {

    override fun configure() {
        val lifecycleService = LifecycleService(logging)
        bind(Config::class.java).toInstance(config)
        bind(LifecycleService::class.java).toInstance(lifecycleService)
        bindListener(LifecycleMatcher(), lifecycleService)
    }
}

